### PR TITLE
Enhance docs for custom locator strategy

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

Clarify how to use the scoped element when using custom selector strategy.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

n/a

### CODE REVIEW

File: website/docs/Selectors.md [hunk 1]
```diff
@@ -505,17 +505,28 @@ browser.react$$('MyComponent') // returns the WebdriverIO Elements for the array
 If your app requires a specific way to fetch elements you can define yourself a custom selector strategy that you can use with `custom$` and `custom$$`. For that register your strategy once in the beginning of the test:
 
 ```js
-browser.addLocatorStrategy('myCustomStrategy', (selector) => {
-    return document.querySelectorAll(selector)
+browser.addLocatorStrategy('myCustomStrategy', (selector, root) => {
```
Comment (erwinheitzman) [inline website/docs/Selectors.md:508]: Can't find anywhere in the code that this should work and when running this I also get a confirmation about this saying that root is undefined. Why is this change made?

**Review by erwinheitzman — CHANGES_REQUESTED** [review]

File: website/docs/Selectors.md (hunk 1 above)
Comment (erwinheitzman) [reply website/docs/Selectors.md:508]: Also, if this is indeed true. We should also update the documentation at all other places and also our types to reflect this as well

Comment (christian-bromann) [reply website/docs/Selectors.md:508]: > Can't find anywhere in the code that this should work and when running this I also get a confirmation about this saying that root is undefined. Why is this change made?

It only works when calling the custom locator strategy on an element object. The reference implementation is here: https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/commands/element/custom$.ts#L37-L51

**Comment by christian-bromann** [comment]
@erwinheitzman I improved the types and added a test.

> Also, if it would be true that root can be used here, we should also update the documentation at all other places and also our types to reflect this as well

I haven't seen an example where we called a custom locator strategy on an element in other places.

**Review by erwinheitzman — APPROVED** [review]

**Pull request merged**